app/actors/hyrax/actors/file_set_actor.rb

```diff
@@ -71,14 +73,15 @@ def create_metadata(file_set_params = {})
       def attach_to_work(work, file_set_params = {})
         acquire_lock_for(work.id) do
           # Ensure we have an up-to-date copy of the members association, so that we append to the end of the list.
-          work.reload unless work.new_record?
+          work = reload(work) unless work.new_record?
           file_set.visibility = work.visibility unless assign_visibility?(file_set_params)
+          save(file_set)
           work.ordered_members << file_set
           work.representative = file_set if work.representative_id.blank?
           work.thumbnail = file_set if work.thumbnail_id.blank?
           # Save the work so the association between the work and the file_set is persisted (head_id)
           # NOTE: the work may not be valid, in which case this save doesn't do anything.
-          work.save
+          save(work)
```

Comment:
This actor needs to be set up to use the parameter `use_valkyrie: false` like was done in https://github.com/samvera/hyrax/blob/094f5284d33252d356eaada1f18e7e1a2d3e3bd6/app/actors/hyrax/actors/base_actor.rb#L22